cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -219,18 +222,36 @@ class DefImpl extends DefOrUseImpl, TDefImpl {
   }
 
   predicate isCertain() { isDef(true, _, address, _, _, ind) }
+
+  /**
+   * Holds if this definition not having a use does not imply that it
+   * will be removed from the SSA def-use relation.
+   */
+  predicate cannotBePruned() { none() }
+
+  /** Gets the definition from the SSA pruning stage that corresponds to this definition. */
+  SsaInternals0::Def getPruningDef() { result.getAddressOperand() = address }
+}
+
+private class DirectDef extends DefImpl, TDirectDef {
+  DirectDef() { this = TDirectDef(address, ind) }
+
+  override int getIndirection() { isDef(_, _, address, _, result, ind) }
+
+  override string toString() { result = "DefImpl" }
```

Comment:
Is `"DefImpl"` still what you want here?

Reply:
Good point. No, that one should just be deleted 👍. Fixed in 6a9aa72550c482ea8732afcef5ac14c8e88f6cd1.

Reply:
Hmmm. Where? I might be suffering from Friday blindness, but I don't see it anywhere.

Reply:
Ah, right. I see what you mean. I'm happy with either solution. For the debugging I've personally done I never really quick eval those classes anyway. However, it should probably be consistent with the `Use`-related classes 😂. 6ed1ef562a69e55b4d1cb0b43222ffaf0c07c93c changes the `toString` on `DirectDef` to be `DirectDef` instead of `DefImpl`. I _think_ they're consistent now.